internal/fleet/agent_policy/delete.go

```diff
@@ -30,6 +34,19 @@ func (r *agentPolicyResource) Delete(ctx context.Context, req resource.DeleteReq
 		return
 	}
 
-	diags = fleet.DeleteAgentPolicy(ctx, client, policyID)
+	// If space_ids is set in state, use space-aware DELETE request
+	if !stateModel.SpaceIds.IsNull() && !stateModel.SpaceIds.IsUnknown() {
+		var tempDiags diag.Diagnostics
+		spaceIDs := utils.ListTypeAs[types.String](ctx, stateModel.SpaceIds, path.Root("space_ids"), &tempDiags)
+		if !tempDiags.HasError() && len(spaceIDs) > 0 {
+			spaceID := spaceIDs[0].ValueString()
+			diags = fleet.DeleteAgentPolicyInSpace(ctx, client, policyID, spaceID)
```

Comment:
Let's assume there are multiple space IDs. 

Does this entirely delete the policy, or does it simply remove the policy from the specified space? From the perspective of an API consumer I would expect the latter, in which case I'd assume we'd need to call this for each space in the slice.